Runtimes/Core/Concurrency/CMakeLists.txt

```diff
@@ -0,0 +1,127 @@
+  Actor.cpp
+  AsyncLet.cpp
+  AsyncStream.cpp
+  Clock.cpp
+  ConcurrencyHooks.cpp
+  EmbeddedSupport.cpp
+  Error.cpp
+  ExecutorChecks.cpp
+  GlobalExecutor.cpp
+  Setup.cpp
+  Task.cpp
+  TaskAlloc.cpp
+  TaskGroup.cpp
+  TaskLocal.cpp
+  TaskStatus.cpp
+  ThreadingError.cpp
+  TracingSignpost.cpp
+  "${PROJECT_SOURCE_DIR}/CompatibilityOverride/CompatibilityOverride.cpp"
+  "${PROJECT_SOURCE_DIR}/linker-support/magic-symbols-for-install-name.c")
+target_sources(swift_Concurrency PRIVATE
+  $<$<STREQUAL:${SwiftCore_CONCURRENCY_GLOBAL_EXECUTOR},dispatch>:DispatchGlobalExecutor.cpp>
+  $<$<STREQUAL:${SwiftCore_CONCURRENCY_GLOBAL_EXECUTOR},singlethreaded>:CooperativeGlobalExecutor.cpp>
+  $<$<STREQUAL:${SwiftCore_CONCURRENCY_GLOBAL_EXECUTOR},hooked>:NonDispatchGlobalExecutor.cpp>
+  $<$<STREQUAL:${SwiftCore_CONCURRENCY_GLOBAL_EXECUTOR},none>:NonDispatchGlobalExecutor.cpp>)
+target_compile_definitions(swift_Concurrency PRIVATE
+  $<$<COMPILE_LANGUAGE:C,CXX>:-DSWIFT_TARGET_LIBRARY_NAME=swift_Concurrency>
+  # NOTE: VS2017 <15.8 would round clamp alignment to alignof(max_align_t) which
+  # was non-conformant. Indicate that we wish to use extended alignment.
+  $<$<AND:$<PLATFORM_ID:Windows>,$<COMPILE_LANGUAGE:C,CXX>>:-D_ENABLE_EXTENDED_ALIGNED_STORAGE=1>
+  $<$<COMPILE_LANGUAGE:C,CXX>:-DSWIFT_CONCURRENCY_USES_DISPATCH=$<STREQUAL:${SwiftCore_CONCURRENCY_GLOBAL_EXECUTOR},dispatch>>
+  $<$<COMPILE_LANGUAGE:C,CXX>:-D__STDC_WANT_LIB_EXT1__=1>)
+target_compile_options(swift_Concurrency PRIVATE
+  $<$<COMPILE_LANGUAGE:Swift>:-parse-stdlib>
+  # NOTE: do not remove until `IsolatedAny` is on by default in all supported
+  # compilers.
+  "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-enable-experimental-feature IsolatedAny>"
+  # NOTE: enable the async frame pointer on Darwin to faciliate debugging.
+  $<$<AND:$<PLATFORM_ID:Darwin>,$<COMPILE_LANGUAGE:C,CXX>>:-fswift-async-fp=always>
+  "$<$<AND:$<PLATFORM_ID:Darwin>,$<COMPILE_LANGUAGE:Swift>>:SHELL:-Xfrontend -swift-async-frame-pointer=always>"
+  "$<$<AND:$<BOOL:${BUILD_SHARED_LIBS}>,$<COMPILE_LANGUAGE:Swift>>:SHELL:-Xcc -Dswift_Concurrency_EXPORTS>"
+  "$<$<AND:$<STREQUAL:${SwiftCore_CONCURRENCY_GLOBAL_EXECUTOR},dispatch>,$<COMPILE_LANGUAGE:Swift>>:-DSWIFT_CONCURRENCY_USES_DISPATCH>"
+  "$<$<AND:$<STREQUAL:${SwiftCore_CONCURRENCY_GLOBAL_EXECUTOR},dispatch>,$<COMPILE_LANGUAGE:Swift>>:SHELL:-Xcc -DSWIFT_CONCURRENCY_USES_DISPATCH>")
+target_include_directories(swift_Concurrency PRIVATE
+  # FIXME: Use of `swift/Runtime/...`, `swift/ABI/...`, and `swift/Demangling/...`
+  "${SwiftCore_SWIFTC_SOURCE_DIR}/include"
+  # FIXME: grant access to `runtime/CMakeConfig.h` which should be available
+  # through the swiftRuntime target.
+  "${PROJECT_BINARY_DIR}/include")
+target_link_libraries(swift_Concurrency PRIVATE
+  swiftShims
+  swiftConcurrencyInternalShims
+  $<$<BOOL:${BUILD_SHARED_LIBS}>:swiftThreading>
+  $<$<STREQUAL:${SwiftCore_CONCURRENCY_GLOBAL_EXECUTOR},dispatch>:dispatch>
```

Comment:
Looking into it, but I'm getting an error here about not finding dispatch on macOS.

Reply:
Weird, that should be part of the system.